messaging/src/test/java/org/axonframework/tracing/IntermediateSpanFactoryTest.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2010-2023. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.tracing;
+
+import org.axonframework.common.AxonConfigurationException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Abstract class test for all intermediate factories that result in a {@link Span}. Contains facilities
+ * to setup tests in a good manner and run them.
+ *
+ * @param <BI> The builder class
+ * @param <SI> The span factory class
+ * @since 4.9.0
+ * @author Mitchell Herrijgers
+ */
+public abstract class IntermediateSpanFactoryTest<BI, SI> {
+    private final SpanFactory spanFactory = new TestSpanFactory();
+
+    /**
+     * Creates a new factory builder that also sets the provided span factory.
+     */
+    protected abstract BI createBuilder(SpanFactory spanFactory);
+
+    /**
+     * Executes the builder method and returns the factory based on the builder.
+     */
+    protected abstract SI createFactoryBasedOnBuilder(BI builder);
+
+    @Test
+    void cannotSetNullSpanFactory() {
+        Assertions.assertThrows(AxonConfigurationException.class, () -> createBuilder(null));
+    }
+
+    /**
+     * Executes the test based on the given parameters.
+     * @param builderAdditions The additions to the builder of the intermediate factory
+     * @param invocation The invocation of the intermediate factory
+     * @param definition The definition of the expected span
+     */
+    protected void test(Function<BI, BI> builderAdditions, Function<SI, Span> invocation, ExpectedSpan definition) {
```

Comment:
What about we add another test method that defaults the `builderAdditions`? Not all tests need to provide additions to the builder to adjust the test.

Reply:
Sounds fair to me, I'll add it.